src/spec.c

```diff
@@ -156,46 +161,34 @@ int isRdbLoading(RedisModuleCtx *ctx) {
 
 IndexSpec *IndexSpec_CreateNew(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
                                QueryError *status) {
+  const char *specName = RedisModule_StringPtrLen(argv[1], NULL);
+  if (dictFetchValue(specDict, specName)) {
+    QueryError_SetCode(status, QUERY_EINDEXEXISTS);
+    return NULL;
+  }
   IndexSpec *sp = IndexSpec_ParseRedisArgs(ctx, argv[1], &argv[2], argc - 2, status);
   if (sp == NULL) {
     return NULL;
   }
 
-  RedisModuleString *keyString = RedisModule_CreateStringPrintf(ctx, INDEX_SPEC_KEY_FMT, sp->name);
-  RedisModuleKey *k = RedisModule_OpenKey(ctx, keyString, REDISMODULE_READ | REDISMODULE_WRITE);
-  RedisModule_FreeString(ctx, keyString);
-
-  // check that the key is empty
-  if (k == NULL || (RedisModule_KeyType(k) != REDISMODULE_KEYTYPE_EMPTY)) {
-    if (RedisModule_ModuleTypeGetType(k) != IndexSpecType) {
-      QueryError_SetCode(status, QUERY_EREDISKEYTYPE);
-    } else {
-      QueryError_SetCode(status, QUERY_EINDEXEXISTS);
-    }
-    IndexSpec_Free(sp);
-    if (k) {
-      RedisModule_CloseKey(k);
-    }
-    return NULL;
-  }
+  dictAdd(specDict, (char *)specName, sp);
 
   sp->uniqueId = spec_unique_ids++;
   // Start the garbage collector
   IndexSpec_StartGC(ctx, sp, GC_DEFAULT_HZ);
 
   CursorList_AddSpec(&RSCursors, sp->name, RSCURSORS_DEFAULT_CAPACITY);
 
-  // set the value in redis
-  RedisModule_ModuleTypeSetValue(k, IndexSpecType, sp);
-  if (sp->flags & Index_Temporary) {
-    RedisModule_SetExpire(k, sp->timeout * 1000);
-  }
+  //  todo : handle expire
+  //  if (sp->flags & Index_Temporary) {
+  //    RedisModule_SetExpire(k, sp->timeout * 1000);
+  //  }
```

Comment:
What about this todo?

Reply:
We still need to do it, I believe it's not a blocker for milestone ..